test/test_dht.cpp

```diff
@@ -2059,5 +2121,103 @@ TORRENT_TEST(routing_table_extended)
 #endif
 }
 
+TORRENT_TEST(read_only_node)
```

Comment:
It could be, I wrote them together because I try to test that the read_only could be set when DHT is running (without restart). I think other tests actually tested DHT without read_only already. split one test without read-only seems doesn't make much sense.
Thought?